baseball/js/app.js

```diff
@@ -0,0 +1,366 @@
+			d3.select('#yearSelect')
+				.on('change', updatePlayers)
+				.selectAll('option')
+				.data(d3.keys(info.years).reverse())
+				.enter()
+				.append('option')
+				.attr('value', function(d){
+					return d;
+				})
+				.text(function(d){
+					return d;
+				});
+
+			var yearSelectValue = d3.select('#yearSelect').property('value');
+
+			//Fill in player select with options
+			d3.select('#playerSelect')
+				.on('change', updatePlayers)
+				.selectAll('option')
+				.data(d3.keys(info.years[yearSelectValue]))
+				.enter()
+				.append('option')
+				.attr('value', function(d){
+					return d;
+				})
+				.text(function(d){
+					return d;
+				});
+
+			var playerSelectValue = d3.select('#playerSelect').property('value');
+			
+			//Create Map
+			map.selectAll('path')
+				.data(mapData.features)
+				.enter()
+				.append('path')
+				.attr('d', path)
+				.attr('fill', '#eef0ef')
+				.attr('stroke', 'rgb(190,190,190)')
+				.attr('stroke-width', 1);
+			
+			//Draw Cities
+			map.selectAll('circle')
+				.data(appData.stadiums)
+				.enter()
+				.append('circle')
+				.classed('stadium', true)
+				.attr('cx', function(d){
+					return projection([d.lon, d.lat])[0];
+				})
+				.attr('cy', function(d){
+					return projection([d.lon, d.lat])[1];
+				})
+				.attr('r', function(d){
+					var radius = d.years[yearSelectValue][playerSelectValue] * 100;
+					return findRadius(radius);
+				})
+				.style('fill', function(d){
+					var radius = d.years[yearSelectValue][playerSelectValue] * 100;
+					return color(findRadius(radius));
+				})
+				.style('opacity', 0.8)
+				.style('cursor', 'pointer')
+				.text(function(d){
+					return d.years[yearSelectValue][playerSelectValue];
+				})
+				.attr('font-size', '14px')
+				.attr('font-weight', 'bold')
+				.attr('fill', '#000')
+				.on('mouseover', function(d){
+				
+					//set stadium name
+					d3.select('.mapTooltip h3').text(function(){
+						return d.name;
+					});
+
+					//set location
+					d3.select('.mapTooltip .location').text(function(){
+						return d.location;
+					});
+
+					//set teamname
+					d3.select('.mapTooltip .teamName').text(function(){
+						return d.teamName;
+					});
+
+					//set capacity
+					d3.select('.mapTooltip .capacity').text(function(){
+						return d.capicity;
+					});
+				
+					//Set batting Average
+					d3.select('.mapTooltip .battingAverage').text(function(){
+						return setbattingAverage(d);
+					});
+
+					d3.select(".mapTooltip")
+						.style("display", "block")
+						.style("left", function() { // stop tooltip from being cutoff at right of page
+						  var right_edge = d3.event.pageX + this.offsetWidth; //this.offsetWidth == width of tooltip
+						  if (right_edge > width) {
+							return (d3.event.pageX - this.offsetWidth)+"px";
+						  }
+						  return (d3.event.pageX)+"px";
+						})
+						.style("top", function() { // stop tooltip from being cutoff at bottom of page
```

Comment:
there's a custom tooltip handler in vizlab...any chance you've looked at that one? are we gonna want to use what's already in vizlab when we get to full vizzies, or is code like what's in this section going to be our best bet?

Reply:
I haven't looked at the vizlab version, I'm sure there's better ways of doing tooltips.  I just am currently mashing tutorials and google searches together to make things work as I learn what works best.